lib/matplotlib/widgets.py

```diff
@@ -1632,34 +1640,41 @@ def __init__(self, canvas, axes, useblit=True, horizOn=False, vertOn=True,
     def connect(self):
         """Connect events."""
         self._cidmotion = self.canvas.mpl_connect('motion_notify_event',
-                                                  self.onmove)
-        self._ciddraw = self.canvas.mpl_connect('draw_event', self.clear)
+                                                  self._onmove)
+        self._ciddraw = self.canvas.mpl_connect('draw_event', self._clear)
 
     def disconnect(self):
         """Disconnect events."""
         self.canvas.mpl_disconnect(self._cidmotion)
         self.canvas.mpl_disconnect(self._ciddraw)
 
+    @_api.deprecated('3.5')
     def clear(self, event):
+        """Clear the cursor."""
+        if self.ignore(event):
+            return
+        self._clear(event)
+        for line in self.vlines + self.hlines:
+            line.set_visible(False)
+
+    def _clear(self, event):
         """Clear the cursor."""
         if self.ignore(event):
             return
         if self.useblit:
             self.background = (
                 self.canvas.copy_from_bbox(self.canvas.figure.bbox))
-        for line in self.vlines + self.hlines:
-            line.set_visible(False)
 
-    def onmove(self, event):
+    onmove = _api.deprecate_privatize_attribute('3.5')
+
+    def _onmove(self, event):
         if self.ignore(event):
             return
         if event.inaxes not in self.axes:
             return
         if not self.canvas.widgetlock.available(self):
             return
         self.needclear = True
-        if not self.visible:
```

Comment:
Why drop this?

Reply:
I think we want to keep this short-circuit to avoid going to `_update` which avoids a `draw_idle` down the line.

Reply:
If that's the case, we might want to clarify that for our future selves with a comment.

Reply:
Well, we want to remove it from here so that the later `.set_visible` is effective. I'll see if we can still skip the `_update`. Alternatively, we should turn the visibility attributes into property setters.

Reply:
Oh, I see now.  We used to _always_ make the lines not visible so if `not self.visible` we could safely bail.

Do we need some more careful logic on this like to call `self._update()` if visible or just made invisible?  I suspect there is now also some path where we can set `self.vertOn` or `self.horizOn` to False and then they never get removed?

Reply:
I re-added a condition on the `_update` calls, which is hopefully sufficient.